### PR TITLE
ESP32 Azure IoT Kit GSG

### PR DESCRIPTION
**Pull request by ryanwinter**
## Purpose
Add the Getting Started Guide for the ESP32 Azure IoT Kit

## Does this introduce a breaking change?
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Follow the GSG.md in the ESP32 Azure IoT Kit project

## What to Check
Verify that the following are valid
* The GSG.md completes successfully

### CODE REVIEW

File: demos/projects/ESPRESSIF/aziotkit/GSG.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+1. Select *Azure IoT middleware for FreeRTOS Main Task Configuration  --->*
+1. Enable *Enable Device Provisioning Sample*
+1. Set the following parameters, using the connection information saved after you created Azure resources.
+
+    Parameter | Value
+    ---------|----------
+     `Azure IoT Device Symmetric Key` | {Your Primary key value}
+     `Azure Device Provisioning Service ID Scope` | {Your ID scope value}
+     `Azure Device Provisioning Service Registration ID` | {Your Device ID value}
+
```
Comment (ewertons) [inline demos/projects/ESPRESSIF/aziotkit/GSG.md:136]:
```suggestion

> The other parameters in the configuration menu are not used in this example, so they do not need to be set.
```

File: demos/projects/ST/b-l475e-iot01a/GSG.md [hunk 2]
```diff
@@ -267,4 +262,4 @@ To remove the entire Azure IoT Central sample application and all its devices an
 
 In this tutorial you built a custom image that contains Azure FreeRTOS middleware sample code, and then flashed the image to the STM DevKit device. You also used the IoT Central portal to create Azure resources, connect the STM DevKit securely to Azure, view telemetry, and send messages.
 
-* For device developers, the suggested next step is to see the other tutorials in the series [Getting started with Azure RTOS](https://go.microsoft.com/fwlink/p/?linkid=2129824).
+* For device developers, the suggested next step is to see the other tutorials in the series [Getting started with FreeRTOS]().
```
Comment (ewertons) [inline demos/projects/ST/b-l475e-iot01a/GSG.md:265]: Is this missing a link?

**Review by ewertons — APPROVED** [review]
Approved with fix suggestions.

File: demos/projects/ST/b-l475e-iot01a/GSG.md (hunk 2 above)
Comment (ryanwinter) [reply demos/projects/ST/b-l475e-iot01a/GSG.md:265]: removed the link, it will be added when the link is ready from the docs page (not available yet)

**Review by wduraes — APPROVED** [review]

**Pull request merged**